lib/bn.js

```diff
@@ -56,7 +56,7 @@
 
   BN.isBN = function isBN (num) {
     return num !== null && typeof num === 'object' &&
-      num.constructor.name === 'BN' && Array.isArray(num.words);
+      num.constructor.name === BN.name && Array.isArray(num.words);
```

Comment:
Yes.

Not entirely sure if it would make sense using a function reference, but it might be handy: ```num.isBN === BN.isBN```

This shouldn't work if `num` is an instance of a different BN - do we want that to work or fail?